### ISSUE TITLE
"No posts." message shown when there actually are posts - should instead say "Posts are hidden." or something and give the option to unhide them after confirming age

### ISSUE THREAD

**Issue by josephrocca** [LemmyNet/lemmy-ui#1860]
### Requirements

- [X] This is a feature request and not a bug report. Otherwise, please create a new [bug report](https://github.com/LemmyNet/lemmy-ui/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=BUG_REPORT.yml) instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this request (or a similar one) already exists.
- [X] It's a single feature. Please don't request multiple features in one issue.

### Describe the feature you'd like

I just tried to visit https://lemmy.world/c/stablediffusion on mobile web (where I'm not logged in) and it shows "No posts." I assumed that it was because there were literally no posts. But it turns out there are posts, it's just that the community has marked itself as NSFW (even though there are no NSFW posts there - but maybe because some future posts could be risque), and apparently the Lemmy UI currently 'lies' about the number of posts in cases like this, and worse - *does not give you an option to show the hidden posts without logging in* (IIUC).

This is a bad user experience, and is actually *worse* than old Reddit because at least with old Reddit you can click a button to show the content (i.e. "I am older than 18 years" button). New Reddit requires you to login, which sucks.

**TL;DR**: If you're not logged in, and the community is marked as NSFW, it should show blurred out posts and show a "Please confirm you're over 18" modal to let you see the content *without logging in*.

**Comment by dessalines** [comment]
This is probably not possible or worth it for UIs, because applying filters to a back end request isn't going to tell you which filter caused there to be no results. If someone can think of some way this would be possible without some heavy SQL penalties, we can re-open.

**Comment by Kojoley** [comment]
How taxing it would be to do what the bug title says: "Posts are hidden" instead of "No posts"? It's shouldn't cost much, isn't it?

**Comment by dessalines** [comment]
Because we have no idea if there are no post results because of post hiding, blocks, no posts yet in that community, etc. Saying "posts are hidden" may be the wrong reason.

**Comment by Kojoley** [comment]
I mean, it shouldn't be costly to check if there are any posts in the community? Something has to be really wrong if it is.

**Comment by dessalines** [comment]
You mean query with every filter variation to figure out which filters are causing them not to show up? Yes, that's very expensive.

**Comment by Kojoley** [comment]
No, just a check if there are posts *without* any filters and if it is > 0 instead of "No posts" show something else.

**Comment by Kojoley** [comment]
From my understanding, the information about posts count in a community always fetched to show in the sidebar (where community statistics are shown), so it should not bring any additional cost even?

**Comment by dessalines** [comment]
> From my understanding, the information about posts count in a community always fetched to show in the sidebar (where community statistics are shown), so it should not bring any additional cost even?

This is the only solution I'd consider, but it still wouldn't work for collections that fetch from multiple communities, ie your front page. It would only work for community-specific pages.

Feel free to open up a PR in lemmy-ui for this.

**Comment by Kojoley** [comment]
> > From my understanding, the information about posts count in a community always fetched to show in the sidebar (where community statistics are shown), so it should not bring any additional cost even?
> 
> This is the only solution I'd consider, but it still wouldn't work for collections that fetch from multiple communities, ie your front page. It would only work for community-specific pages.
> 
> Feel free to open up a PR in lemmy-ui for this.

Here it is https://github.com/LemmyNet/lemmy-ui/issues/2536.

**Comment by dessalines** [comment]
You didn't open a PR, but another issue with the same text.
